flink-core/src/main/java/org/apache/flink/configuration/YamlParserUtils.java

```diff
@@ -0,0 +1,227 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Map;
+
+/**
+ * This class contains utility methods to load standard yaml file and convert object to standard
+ * yaml syntax.
+ */
+public class YamlParserUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(YamlParserUtils.class);
+
+    private static final Yaml yaml;
+
+    private static final DumperOptions dumperOptions = new DumperOptions();
+
+    private static final LoaderOptions loaderOptions = new LoaderOptions();
+
+    static {
+        // Make the dump output is in single line
+        dumperOptions.setDefaultFlowStyle(DumperOptions.FlowStyle.FLOW);
+        dumperOptions.setWidth(Integer.MAX_VALUE);
+        // The standard YAML do not allow duplicate keys.
+        loaderOptions.setAllowDuplicateKeys(false);
+
+        yaml =
+                new Yaml(
+                        new Constructor(loaderOptions),
+                        new FlinkConfigRepresenter(dumperOptions),
+                        dumperOptions,
+                        loaderOptions);
+    }
+
+    public static synchronized Map<String, Object> loadYamlFile(File file) throws Exception {
+        try (FileInputStream inputStream = new FileInputStream((file))) {
+            return yaml.load(inputStream);
+        } catch (FileNotFoundException e) {
+            LOG.error("Failed to find YAML file", e);
+            throw e;
+        } catch (IOException | YAMLException e) {
+            if (e instanceof MarkedYAMLException) {
+                YAMLException exception =
+                        wrapExceptionToHiddenSensitiveData((MarkedYAMLException) e);
+                LOG.error("Failed to parse YAML configuration", exception);
+                throw exception;
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * Converts the given value to a string representation in the YAML syntax for configuration
+     * data.
+     *
+     * <p>NOTE: It will skip escaping string values in order to avoid ambiguity.
+     *
+     * @param value The value to be converted.
+     * @return The string representation of the value in YAML syntax.
+     */
+    public static String toConfigurationDataString(Object value) {
+        if (value instanceof String) {
+            return value.toString();
+        } else {
+            return toYAMLString(value);
+        }
+    }
+
+    /**
+     * Converts the given value to a string representation in the YAML syntax. This method uses a
+     * YAML parser to convert the object to YAML format.
+     *
+     * <p>The resulting YAML string may have line breaks at the end of each line. This method
+     * removes the line break at the end of the string if it exists.
+     *
+     * <p>Note: This method may perform escaping on certain characters in the value to ensure proper
+     * YAML syntax.
```

Comment:
It is a bit confusing to me because here it says it may perform escaping. However, in `toConfigurationDataString()` which uses this method it says it skips escaping.